<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BandwidthTestDialog</class>
 <widget class="QDialog" name="BandwidthTestDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>350</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Bandwidth Test Configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="titleLabel">
     <property name="text">
      <string>Configure Bandwidth Test Parameters</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="styleSheet">
      <string>QLabel { font-weight: bold; font-size: 12pt; }</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="parametersGroupBox">
     <property name="title">
      <string>Test Parameters</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::ExpandingFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="targetLabel">
        <property name="text">
         <string>Target IP:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="targetLineEdit">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="portLabel">
        <property name="text">
         <string>Port:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="portSpinBox">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>5201</number>
        </property>
        <property name="toolTip">
         <string>Server port (default: 5201 for iperf3)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="durationLabel">
        <property name="text">
         <string>Duration (seconds):</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="durationSpinBox">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>60</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
        <property name="toolTip">
         <string>Test duration (1-60 seconds, longer = more accurate)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="protocolLabel">
        <property name="text">
         <string>Protocol:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QComboBox" name="protocolComboBox">
        <property name="toolTip">
         <string>TCP (recommended for real-world) or UDP (maximum throughput)</string>
        </property>
        <item>
         <property name="text">
          <string>TCP</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>UDP</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="directionLabel">
        <property name="text">
         <string>Direction:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QComboBox" name="directionComboBox">
        <property name="toolTip">
         <string>Download (receive from server) or Upload (send to server)</string>
        </property>
        <item>
         <property name="text">
          <string>Download</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Upload</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="packetSizeLabel">
        <property name="text">
         <string>Packet Size (KB):</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QSpinBox" name="packetSizeSpinBox">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1024</number>
        </property>
        <property name="value">
         <number>64</number>
        </property>
        <property name="suffix">
         <string> KB</string>
        </property>
        <property name="toolTip">
         <string>Packet size (1-1024 KB, larger = less overhead)</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="infoLabel">
     <property name="text">
      <string>⚠️ Note: Requires custom LanScan server on target device (under development).
Not compatible with iperf3 or standard tools.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="styleSheet">
      <string>QLabel { font-style: italic; padding: 10px; }</string>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>BandwidthTestDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>329</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>174</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>BandwidthTestDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>224</x>
     <y>329</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>174</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
